src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
+    {
+        protected readonly PyObject pyObject;
+
+        public IterableWrapper(PyObject pyObj)
+        {
+            if (pyObj == null)
+                throw new ArgumentNullException();
+            pyObject = pyObj;
```

Comment:
I believe you should make a new `PyObject` here, pointing to the same object in Python (e.g. `pyObject = new PyObject(pyObj.Reference)`). The reason is that if the original `PyObject` is disposed (as suggested below in `ListWrapper.this[int index]`), this iterator will become useless.

Reply:
@lostmsu thanks, I often forget that C# objects can be explicitly disposed before they get axed by the GC.